saleor/graphql/webhook/mutations.py

```diff
@@ -301,3 +316,166 @@ def perform_mutation(cls, _root, info, **data):
+
+
+class WebhookDryRun(BaseMutation):
+    payload = JSONString(
+        description="JSON payload, that would be sent out to webhook's target URL."
+    )
+
+    class Arguments:
+        query = graphene.String(
+            description="The subscription query that defines the webhook event and its "
+            "payload.",
+            required=True,
+        )
+        object_id = graphene.ID(
+            description="The ID of an object to serialize.", required=True
+        )
+
+    class Meta:
+        description = "Performs a dry run of a webhook event."
+        permissions = (AppPermission.MANAGE_APPS,)
+        error_type_class = WebhookDryRunError
+
+    @classmethod
+    def validate_input(cls, info, **data):
+        query = data.get("query")
+        object_id = data.get("object_id")
+
+        event_type = get_event_type_from_subscription(query)
+        if not event_type:
+            raise_validation_error(
+                field="query",
+                message="Can't parse an event type from query.",
+                code=WebhookDryRunErrorCode.GRAPHQL_ERROR,
+            )
+
+        event = WEBHOOK_TYPES_MAP.get(event_type)
+        if not event:
+            event_name = event_type[0].upper() + to_camel_case(event_type)[1:]
+            raise_validation_error(
+                field="query",
+                message=f"Event type: {event_name} is not defined in graphql schema.",
+                code=WebhookDryRunErrorCode.GRAPHQL_ERROR,
+            )
+
+        model, _ = graphene.Node.from_global_id(object_id)
+        if model != WebhookEventAsyncType.ROOT_TYPE.get(event_type):
+            raise_validation_error(
+                field="objectId",
+                message="ObjectId doesn't match event type.",
+                code=WebhookDryRunErrorCode.INVALID_ID,
+            )
+
+        object = cls.get_node_or_error(info, object_id, field="objectId")
+
+        if permission := WebhookEventAsyncType.PERMISSIONS.get(event_type):
+            codename = permission.value.split(".")[1]
+            user_permissions = [
+                perm.codename for perm in info.context.user.effective_permissions.all()
+            ]
+            if codename not in user_permissions:
+                raise_validation_error(
+                    message=f"The user doesn't have required permission: {codename}.",
+                    code=WebhookDryRunErrorCode.MISSING_PERMISSION,
+                )
+
+        return event_type, object, query
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        event_type, object, query = cls.validate_input(info, **data)
+        payload = None
+        if all([event_type, object, query]):
+            request = initialize_request()
+            payload = generate_payload_from_subscription(
+                event_type, object, query, request
+            )
+        return WebhookDryRun(payload=payload)
+
+
+class WebhookTrigger(BaseMutation):
```

Comment:
Are we going to release it as part of this PR or a separate one? It would be better to move it to some other branch.